src/internal.rs

```diff
@@ -367,6 +379,15 @@ pub struct Rav1dContext {
     pub(crate) picture_pool: *mut Rav1dMemPool,
 }
 
+impl Rav1dContext {
+    /// Iterates over all frame contexts in the `fc` buffer, starting at a given
+    /// index. The iterator wraps around to the start of the buffer. so all
+    /// contexts will be included.
+    pub(crate) fn fc_iter(&self, start: usize) -> impl Iterator<Item = &Rav1dFrameContext> {
+        self.fc.iter().skip(start).chain(self.fc.iter().take(start))
+    }
```

Comment:
Maybe it should be a `VecDeque`?  This should be fine for now, though.

Reply:
I don't really want it to be a Vec at all, it's fixed-size after initialization. I think I'd rather have this bit of weirdness, we only wrap in a couple of places.